Add generators

tchiadeu opened the pull request:
Here is a setup to add a small generator.
Maybe the naming is a little bit redundant but it's to keep in mind than the form object will be in forms folder.

If you merge the PR, you can also add the information in the readme.

tchiadeu committed:
organize directly in generators folder

tchiadeu committed:
add form generator

tchiadeu committed:
hook minitest for form generator

tchiadeu committed:
hook rspec for form generator

Intrepidd reviewed (changes requested):
Thanks ! That's a great contribution :) 

Any way to have a more short generator experience ? maybe `rails g form foo_bar` instead of `rails g hyper_active_form:form`

I'm not sure this is even possible, I don't have much experience with rails generators.

Also, when trying it out, it generated a test unit spec although the repo I tried it in uses rspec, any idea why ?

tchiadeu committed:
shorter experience to generate form

you can now go for "rails g form foo_bar" instead of "rails g
hyperactiveform:form foo_bar"

tchiadeu commented:
Maybe you didn't changed your test framework ?

You have to do something like :
```ruby
# config/application.rb

config.generators do |generate|
  generate.test_framework :rspec
end
```

Most of the time, I go for this config (for information) :
```ruby
# config/application.rb

config.generators do |generate|
  generate.assets false
  generate.helper false
  generate.test_framework :rspec,
                                              :fixture_replacement: :factory_bot,
                                              :helper_specs: false,
                                              :view_specs: false,
                                              :routing_specs: false
  generate.fixture_replacement :factory_bot, dir: 'spec/factories'
end
```

You can have a shorter experience to generate files via `rails g form foo_bar`, I change the files to do so.

Intrepidd commented:
Thanks a lot ❤️ 

Released v0.3.0 with this change : https://github.com/Intrepidd/hyperactiveform/releases/tag/v0.3.0
